Model/OrderImporter/AllegroReservation.php

```diff
@@ -0,0 +1,244 @@
+    private $placeReservationsForSalesEvent;
+
+    /** @var StoreManagerInterface */
+    private $storeManager;
+
+    /** @var ReservationRepository */
+    private $reservationRepository;
+
+    /** @var ResourceConnection */
+    private $resource;
+
+    /** @var Configuration */
+    private $configuration;
+
+    /**
+     * AllegroReservation constructor.
+     * @param ProductRepositoryInterface $productRepository
+     * @param ItemToSellInterfaceFactory $itemsToSellFactory
+     * @param SalesEventInterfaceFactory $salesEventFactory
+     * @param SalesChannelInterfaceFactory $salesChannelFactory
+     * @param PlaceReservationsForSalesEventInterface $placeReservationsForSalesEvent
+     * @param StoreManagerInterface $storeManager
+     * @param ReservationRepository $reservationRepository
+     * @param ResourceConnection $resource
+     * @param Configuration $configuration
+     */
+    public function __construct(
+        ProductRepositoryInterface $productRepository,
+        ItemToSellInterfaceFactory $itemsToSellFactory,
+        SalesEventInterfaceFactory $salesEventFactory,
+        SalesChannelInterfaceFactory $salesChannelFactory,
+        PlaceReservationsForSalesEventInterface $placeReservationsForSalesEvent,
+        StoreManagerInterface $storeManager,
+        ReservationRepository $reservationRepository,
+        ResourceConnection $resource,
+        Configuration $configuration
+    ) {
+        $this->productRepository = $productRepository;
+        $this->itemsToSellFactory = $itemsToSellFactory;
+        $this->salesEventFactory = $salesEventFactory;
+        $this->salesChannelFactory = $salesChannelFactory;
+        $this->placeReservationsForSalesEvent = $placeReservationsForSalesEvent;
+        $this->storeManager = $storeManager;
+        $this->reservationRepository = $reservationRepository;
+        $this->resource = $resource;
+        $this->configuration = $configuration;
+    }
+
+    /**
+     * @param CheckoutFormInterface $checkoutForm
+     * @throws \Exception
+     */
+    public function placeReservation(CheckoutFormInterface $checkoutForm): void
+    {
+        try {
+            if (!$this->configuration->areReservationsEnabled()) {
+                return;
+            }
+            $this->placeReservationsForSalesEvent->execute(
+                $this->getItemsToSell($checkoutForm, false),
+                $this->getSalesChannel(),
+                $this->getSalesEvent(self::ALLEGRO_EVENT_RESERVATION_PLACED, $checkoutForm->getId()));
+        } catch (\Exception $e) {
+            throw new \Exception(
+                "Error while placing reservation for order with id [{$checkoutForm->getId()}]",
+                0,
+                $e
+            );
+        }
+    }
+
+    /**
+     * @param CheckoutFormInterface $checkoutForm
+     * @throws \Exception
+     */
+    public function compensateReservation(CheckoutFormInterface $checkoutForm): void
```

Comment:
This method looks 100% like placeReservation witch seems like a bug or repetition, please check.